### PR TITLE
Add default subject to Collections

### PR DESCRIPTION
**Pull request by zwolf**
Collections now have a default subject field. It's serialized as either the media's first location's URL or just the collection's first subject's if nil. This creates an extra db query for the medium but prevents an entire extra API call from PFE for the first linked subject just to get its URL.

I need to test this on staging with the JS client and make sure it's set up to handle however it builds requests (via `/api/collections/:id` and `/api/collections/:id/links/default_subject`)

# Review checklist

- [ ] First, the most important one: is this PR small enough that you can actually review it? Feel free to just reject a branch if the changes are hard to review due to the length of the diff.
- [ ] If there are any migrations, will they the previous version of the app work correctly after they've been run (e.g. the don't remove columns still known about by ActiveRecord).
- [ ] If anything changed with regards to the public API, are those changes also documented in the `apiary.apib` file?
- [ ] Are all the changes covered by tests? Think about any possible edge cases that might be left untested.

### CODE REVIEW

File: db/migrate/20170403194826_add_default_subject_to_collections.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddDefaultSubjectToCollections < ActiveRecord::Migration
+  def change
+    add_column :collections, :default_subject_id, :integer
```
Comment (marten) [inline db/migrate/20170403194826_add_default_subject_to_collections.rb]: Should we get a foreign key on this too?

File: apiary.apib [hunk 2]
```diff
@@ -3047,6 +3050,33 @@ owner or have edit permissions.
 
 + Response 204
 
+## Add default subject link [/collections/{id}/links/default_subject]
+Links a default subject to a collection. This subject's first media
+location URL will be included in the serialized collection and used
+as the thumbnail. Update this attribute with `null` to use the first
```
Comment (marten) [inline apiary.apib]: Should the backend be taking care of that default? It doesn't seem to be in this PR.

Comment (zwolf) [inline apiary.apib]: This? https://github.com/zwolf/Panoptes/blob/6a1f7806ad2c5bf2e4a0de2b8c98d9ae7ec98d56/app/serializers/collection_serializer.rb#L26

File: db/migrate/20170403194826_add_default_subject_to_collections.rb (hunk 1 above)
Comment (zwolf) [inline db/migrate/20170403194826_add_default_subject_to_collections.rb]: ✔️

File: apiary.apib (hunk 2 above)
Comment (zwolf) [inline apiary.apib]: Or if you mean the route, it's handled here

https://github.com/zooniverse/Panoptes/pull/2296/files#diff-21497849d8f00507c9c8dcaf6288b136R124

and setting is allowed via this:
https://github.com/zooniverse/Panoptes/pull/2296/files#diff-e5f000780a616ca07cd53cdbbfdbfd59R14

Comment (marten) [inline apiary.apib]: Yeah like that serializer. Missed that, 👍

**Pull request merged**